lib/commons/aria/is-combobox-popup.js

```diff
@@ -0,0 +1,55 @@
+import getRole from './get-role';
+import ariaAttrs from '../../standards/aria-attrs';
+import { getRootNode } from '../../core/utils';
+
+/**
+ * Whether an element is the popup for a combobox
+ * @method isComboboxPopup
+ * @memberof axe.commons.aria
+ * @instance
+ * @param {VirtualNode} virtualNode
+ * @param {Object} optionsw
+ * @property {String[]} popupRoles Overrides which roles can be popup. Defaults to aria-haspopup values
+ * @returns {boolean}
+ */
+export default function isComboboxPopup(virtualNode, { popupRoles } = {}) {
+  const role = getRole(virtualNode);
+  popupRoles ??= ariaAttrs['aria-haspopup'].values;
+  if (!popupRoles.includes(role)) {
+    return false;
+  }
+
+  // in ARIA 1.1 the container has role=combobox
+  const vParent = nearestParentWithRole(virtualNode);
+  if (isCombobox(vParent)) {
+    return true;
+  }
+
+  const { id } = virtualNode.props;
+  if (!id) {
+    return false;
+  }
+
+  if (!virtualNode.actualNode) {
+    throw new Error('Unable to determine combobox popup without an actualNode');
```

Comment:
I didn't think we needed additional virtualNode tests, but let me know if I missed anything.